### PR TITLE
feat(api,hatchery): allow hatchery to handle jobrun

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18936.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by richardlt — APPROVED** [review]

File: sdk/websocket.go [hunk 1]
```diff
@@ -127,3 +132,19 @@ type WebsocketEvent struct {
...
+}
+
+type WebsocketJobQueueEvent struct {
+	JobRunID     string
+	Region       string
+	ModelType    string
+	ProjectKey   string
+	WorkflowName string
+	JobID        string
+	RunNumber    int64
```
Comment (yesnault) [inline sdk/websocket.go:149]:
```suggestion
	JobRunID     string `json:"job_run_id"`
	Region       string `json:"region"`
	ModelType    string `json:"model_type"`
	ProjectKey   string `json:"project_key"`
	WorkflowName string `json:"workflow_name"`
	JobID        string `json:"job_id"`
	RunNumber    int64 `json:"run_number"`
```

**Review by yesnault — CHANGES_REQUESTED** [review]

File: engine/api/workflow_v2/dao_run_job_info.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"time"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/engine/gorpmapper"
+	"github.com/ovh/cds/sdk"
+)
+
+func getAllRunJobInfos(ctx context.Context, db gorp.SqlExecutor, query gorpmapping.Query) ([]sdk.V2WorkflowRunJobInfo, error) {
```
Comment (fsamin) [inline engine/api/workflow_v2/dao_run_job_info.go:14]: Information is uncountable in english

**Review by fsamin — CHANGES_REQUESTED** [review]

File: engine/api/workflow_v2/dao_run_job_info.go (hunk 2 above)
Comment (sguiheux) [reply engine/api/workflow_v2/dao_run_job_info.go:14]: done

File: engine/api/v2_workflow_run_engine.go [hunk 3]
```diff
@@ -154,12 +175,29 @@ func (api *API) workflowRunV2Trigger(ctx context.Context, wrEnqueue sdk.V2Workfl
 		}
 	}
 
+	// Send to websocket
```
[the remark this answers is not in the document]
Comment (sguiheux) [reply engine/api/v2_workflow_run_engine.go:178]: done

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18943.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18945.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✔

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18951.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18970.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Pull request merged**